server/integration/testsuite/src/test/java/org/infinispan/server/test/partitionhandling/PartitionHandlingIT.java

```diff
@@ -0,0 +1,252 @@
+
+   @InfinispanResource(CONTAINER2)
+   RemoteInfinispanServer server2;
+
+   @InfinispanResource(CONTAINER3)
+   RemoteInfinispanServer server3;
+   String server3Address;
+
+   @Before
+   public void setUp() {
+      server1Address = server1.getHotrodEndpoint().getInetAddress().getHostAddress() + ":" + server1.getHotrodEndpoint().getPort();
+      server3Address = server3.getHotrodEndpoint().getInetAddress().getHostAddress() + ":" + server3.getHotrodEndpoint().getPort();
+
+      if (partitionHandlingController == null) {
+         String node0Ip = System.getProperty("node0.ip");
+         String node1Ip = System.getProperty("node1.ip");
+         String node2Ip = System.getProperty("node2.ip");
+         int node0Port = Integer.valueOf(System.getProperty("node0.mgmt.port"));
+         int node1Port = Integer.valueOf(System.getProperty("node1.mgmt.port"));
+         int node2Port = Integer.valueOf(System.getProperty("node2.mgmt.port"));
+
+         StandaloneManagementClient[] clients = new StandaloneManagementClient[3];
+         clients[0] = new StandaloneManagementClient(node0Ip, node0Port, "node0");
+         clients[1] = new StandaloneManagementClient(node1Ip, node1Port, "node1");
+         clients[2] = new StandaloneManagementClient(node2Ip, node2Port, "node2");
+         partitionHandlingController = new PartitionHandlingController(clients);
+      }
+   }
+
+   /**
+    * We use a cache with owners=1, otherwise the bigger partition can always handle all writes.
+    */
+   @Test
+   public void testDenyReadWrites() throws InterruptedException {
+      RemoteCacheManager cacheManager = ITestUtils.createCacheManager(server2);
+      RemoteCache<Object, Object> cache = cacheManager.getCache(DENY_READ_WRITES_CACHE);
+
+      String server1OwnedKey = getKeyOwnedByNode(server1Address, cacheManager, cache, 1);
+      String server3OwnedKey = getKeyOwnedByNode(server3Address, cacheManager, cache, 1);
+
+      partitionCluster();
+
+      RemoteCache<Object, Object> denyReadWritesCache1 = ITestUtils.createCacheManager(server1).getCache(DENY_READ_WRITES_CACHE);
+      RemoteCache<Object, Object> denyReadWritesCache3 = ITestUtils.createCacheManager(server3).getCache(DENY_READ_WRITES_CACHE);
+
+      denyReadWritesCache1.put(server1OwnedKey, "value");
+      assertEquals("value", denyReadWritesCache1.get(server1OwnedKey));
+      assertThrowsHotRodClientException(() -> denyReadWritesCache1.put(server3OwnedKey, "value"));
+      assertThrowsHotRodClientException(() -> denyReadWritesCache1.get(server3OwnedKey));
+
+      denyReadWritesCache3.put(server3OwnedKey, "value");
+      assertEquals("value", denyReadWritesCache3.get(server3OwnedKey));
+      assertThrowsHotRodClientException(() -> denyReadWritesCache3.put(server1OwnedKey, "value"));
+      assertThrowsHotRodClientException(() -> denyReadWritesCache3.get(server1OwnedKey));
+
+      healCluster();
+   }
+
+   @Test
+   public void testAllowReads() throws InterruptedException {
```

Comment:
Other than the cache name, the content of `testAllowReads` and `testDenyReadWrites` is the same. We should just create one method with this logic that takes the cache name as a param and simply call this in the test methods with the appropriate name.